engine/table/src/main/java/io/deephaven/engine/util/TableTools.java

```diff
@@ -665,8 +665,9 @@ public static ColumnHolder<Float> floatCol(String name, float... data) {
      * @return a Deephaven Table with no columns.
      */
     public static Table emptyTable(long size) {
-        return new QueryTable(RowSetFactory.flat(size).toTracking(),
-                Collections.emptyMap());
+        final QueryTable result = new QueryTable(RowSetFactory.flat(size).toTracking(), Collections.emptyMap());
+        result.setFlat();
+        return result;
```

Comment:
This doesn't matter, but sometimes what we do to avoid the intermediate variables looks like:
```
    return new QueryTable(rowSet, visibleSources) {
            {
                setFlat();
            }
```